implementations/rust/ockam/ockam/src/remote/lifecycle.rs

```diff
@@ -0,0 +1,224 @@
+use crate::remote::{Addresses, RemoteForwarder, RemoteForwarderInfo, RemoteForwarderTrustOptions};
+use crate::Context;
+use core::time::Duration;
+use ockam_core::compat::sync::Arc;
+use ockam_core::compat::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use ockam_core::{
+    route, Address, AllowAll, AllowSourceAddress, DenyAll, Mailbox, Mailboxes,
+    OutgoingAccessControl, Result, Route,
+};
+use ockam_node::{DelayedEvent, WorkerBuilder};
+use tracing::debug;
+
+#[derive(Clone, Copy)]
+pub(super) enum Ftype {
+    Static,
+    Ephemeral,
+    StaticWithoutHeartbeats,
+}
+
+impl Ftype {
+    pub fn str(&self) -> &'static str {
+        match self {
+            Ftype::Static => "static",
+            Ftype::Ephemeral => "ephemeral",
+            Ftype::StaticWithoutHeartbeats => "static_w/o_heartbeats",
+        }
+    }
+}
+
+impl RemoteForwarder {
+    fn mailboxes(
+        addresses: Addresses,
+        heartbeat_source_address: Option<Address>,
+        outgoing_access_control: Arc<dyn OutgoingAccessControl>,
+    ) -> Mailboxes {
+        let main_internal = Mailbox::new(
+            addresses.main_internal,
+            Arc::new(DenyAll),
+            outgoing_access_control,
+        );
+
+        let main_remote = Mailbox::new(
+            addresses.main_remote,
+            Arc::new(AllowAll),
+            Arc::new(AllowAll),
+        );
+
+        let mut additional_mailboxes = vec![main_remote];
+
+        if let Some(heartbeat_source_address) = heartbeat_source_address {
+            let heartbeat = Mailbox::new(
+                addresses.heartbeat,
+                Arc::new(AllowSourceAddress(heartbeat_source_address)),
+                Arc::new(DenyAll),
+            );
+            additional_mailboxes.push(heartbeat);
+        }
+
+        Mailboxes::new(main_internal, additional_mailboxes)
+    }
+}
+
+impl RemoteForwarder {
+    fn new(
+        addresses: Addresses,
+        registration_route: Route,
+        registration_payload: String,
+        heartbeat: Option<DelayedEvent<Vec<u8>>>,
+        heartbeat_interval: Duration,
+    ) -> Self {
+        Self {
+            addresses,
+            completion_msg_sent: false,
+            registration_route,
+            registration_payload,
+            heartbeat,
+            heartbeat_interval,
+        }
+    }
+
+    /// Create and start static RemoteForwarder at predefined address with given Ockam Hub route
+    pub async fn create_static(
+        ctx: &Context,
+        hub_route: impl Into<Route>,
```

Comment:
We probably need to do a full pass on `Hub`. There are many places where it is still used.

Reply:
https://github.com/build-trust/ockam/issues/4520